Removing magic constant from template param check

SecMeant opened the pull request:
Storing digits in variable, thus removing magic constant from template param check.

SecMeant committed:
Storing digits in variable, thus removing magic constant from template param check

tcsullivan commented:
This would be a nice improvement; however, it's leaving us with `digits` and `digit_count` in the global scope, which could mess up other code. One solution to that would be to put everything in a namespace, though I haven't done that yet (not sure what to name it). Open to any other solutions if you have them.

Neargye commented:
@tcsullivan `namespace cits // constexpr_integral_to_string` ?

Neargye commented on to_string.hpp:
```diff
@@ -9,6 +9,9 @@
 
 #include <type_traits>
 
+constexpr char digits[] = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ";
```
`inline constexpr `

m-peko replied:
@Neargye why `inline constexpr` when `constexpr` itself is implicitly inline?

Neargye replied:
​constexpr specifier implies  inline for static data members as well as functions, but not for global variable. 
https://en.cppreference.com/w/cpp/language/inline

Neargye commented on to_string.hpp:
```diff
@@ -9,6 +9,9 @@
 #include <type_traits>
 
+constexpr char digits[] = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+constexpr auto digit_count = sizeof(digits) / sizeof(digits[0]);
```
also `inline constexpr`

tcsullivan committed:
Merge branch 'master' into digits_array

SecMeant committed:
Adding constexpr_to_string namespace

Namespace is used to prevent pollution of global namespace
by digits and digits_count symbols.

SecMeant commented:
Now, when I think about it, maybe we should use something like

namespace to_string_ { ... }

Since its only purpose is to only hide digits symbols.
@tcsullivan

tcsullivan commented:
@SecMeant The `constexpr_to_string` namespace is honestly probably enough. I just realized that the final `to_string` declaration is outside of the namespace, so this isn't actually affecting how `to_string` is called (something I was concerned about).

I'm good to merge this PR as it is now, if you are.

SecMeant commented:
Sure
